libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyManagerTests.java

```diff
@@ -209,7 +238,10 @@ public void testRequestingModuleWithStackWalk() throws IOException, ClassNotFoun
         var instrumentedClass = makeClassInItsOwnModule();    // The class that called the check method
         var ignorableClass = makeClassInItsOwnModule();
 
-        var policyManager = policyManagerWithEntitlementsModule(entitlementsClass.getModule());
+        var policyManager = policyManagerWithEntitlementsModule(
+            ClassLoader.getSystemClassLoader().getUnnamedModule(),
```

Comment:
Instead of relying on the real classloader, can we create a test class in its own loader, similar to the above `makeClassInItsOwnModule` above? Then we can test that class gets the right policy when another class in an unnamed module does not.

Reply:
Added a test for a test agents module.